### PR TITLE
fix #3153 link shortening on post previews

### LINKED ISSUE #3153: Bring back link shortening feature
Users were used to share short links of their posts via the "Share" button on the Post preview screen (prior 4.4). We should reintroduce the short link shortening when a user share a post via the "new" preview screen.

Note: use`WPMeShortlinks`

### PR DESCRIPTION
**Pull request by maxme**
fix #3153 link shortening on post previews

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/ActivityLauncher.java [hunk 1]
```diff
@@ -161,12 +161,15 @@ public static void browsePostOrPage(Context context, Blog blog, Post post) {
         if (blog == null || post == null || TextUtils.isEmpty(post.getPermaLink())) return;
 
         String url = post.getPermaLink();
-        if (-1 == url.indexOf('?')) {
-            url = url.concat("?preview=true");
-        } else {
-            url = url.concat("&preview=true");
+        // Add the preview=true parameter if the post is not published yet
+        if (!post.isPublished()) {
```
Comment (maxme) [inline WordPress/src/main/java/org/wordpress/android/ui/ActivityLauncher.java]: Actually, I should check "if post has non published change", not sure if that state exists.

Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/ui/ActivityLauncher.java]: You should be able to rely on `post.isLocalChange()` for that.

Comment (maxme) [inline WordPress/src/main/java/org/wordpress/android/ui/ActivityLauncher.java]: Hm, I would like a "isRemoteChangeNotPublished()" ;) - I think that's actually not possible

File: WordPress/src/main/java/org/wordpress/android/ui/ActivityLauncher.java [hunk 2]
```diff
@@ -161,12 +161,15 @@ public static void browsePostOrPage(Context context, Blog blog, Post post) {
...
+            if (-1 == url.indexOf('?')) {
```
Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/ui/ActivityLauncher.java]: We really should have a `WordPressUtils` function that appends the URL param.

File: WordPress/src/main/java/org/wordpress/android/ui/WPWebViewActivity.java [hunk 3]
```diff
@@ -78,19 +81,19 @@ public static void openUrlByUsingWPCOMCredentials(Context context, String url, S
         openWPCOMURL(context, url, user);
     }
 
-    public static void openUrlByUsingBlogCredentials(Context context, Blog blog, String url) {
+    public static void openUrlByUsingBlogCredentials(Context context, Blog blog, Post post, String url) {
         if (context == null) {
-            AppLog.e(AppLog.T.UTILS, "Context is null!!!");
```
Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/ui/WPWebViewActivity.java]: I hear the screams of a thousand exclamation points dying.

**Comment by nbradbury** [comment]
I just realized that `browsePostOrPage` won't be used for unpublished posts (ie: local drafts). That method is only called from the "View" icon in the post list, and that icon doesn't appear for local drafts.

File: WordPress/src/main/java/org/wordpress/android/ui/ActivityLauncher.java (hunk 2 above)
Comment (maxme) [inline WordPress/src/main/java/org/wordpress/android/ui/ActivityLauncher.java]: > We really should have a WordPressUtils function that appends the URL param.

Moved to UrlUtils.

**Comment by nbradbury** [comment]
Looks good :shipit:

**Pull request merged**